src/components/EditCoursePage/EditCourseForm.jsx

```diff
@@ -203,30 +172,41 @@ export class BaseEditCourseForm extends React.Component {
       title,
       pristine,
       uuid,
+      type,
       courseInReview,
       courseStatuses,
       id,
       isSubmittingForReview,
       editable,
       courseInfo,
       reset,
+      courseOptions,
+      courseRunOptions,
     } = this.props;
     const {
       open,
     } = this.state;
 
-    const courseOptions = this.getCourseOptions();
-    const courseRunOptions = this.getCourseRunOptions();
-    const levelTypeOptions = courseOptions && this.parseOptions(courseOptions.level_type.choices);
-    const ofacRestrictionOptions = (courseRunOptions && courseRunOptions.has_ofac_restrictions &&
-      this.parseOptions(courseRunOptions.has_ofac_restrictions.choices));
-    const subjectOptions = courseOptions && this.parseOptions(courseOptions.subjects.child.choices);
-    const pacingTypeOptions = (courseRunOptions &&
-      this.parseOptions(courseRunOptions.pacing_type.choices));
-    const languageOptions = (courseRunOptions &&
-      this.parseOptions(courseRunOptions.content_language.choices));
-    const programOptions = (courseRunOptions &&
-      this.parseOptions(courseRunOptions.expected_program_type.choices));
+    const courseOptionsData = getOptionsData(courseOptions);
+    const courseRunOptionsData = getOptionsData(courseRunOptions);
+    const levelTypeOptions = courseOptionsData &&
+      this.parseOptions(courseOptionsData.level_type.choices);
+    const ofacRestrictionOptions = (courseRunOptionsData &&
+      courseRunOptionsData.has_ofac_restrictions &&
+      this.parseOptions(courseRunOptionsData.has_ofac_restrictions.choices));
+    const subjectOptions = courseOptionsData &&
+      this.parseOptions(courseOptionsData.subjects.child.choices);
+    const pacingTypeOptions = (courseRunOptionsData &&
+      this.parseOptions(courseRunOptionsData.pacing_type.choices));
+    const languageOptions = (courseRunOptionsData &&
+      this.parseOptions(courseRunOptionsData.content_language.choices));
+    const programOptions = (courseRunOptionsData &&
+      this.parseOptions(courseRunOptionsData.expected_program_type.choices));
+    const parsedTypeOptions = courseOptionsData &&
+      parseCourseTypeOptions(courseOptionsData.type.type_options);
+    const { courseTypeOptions } = parsedTypeOptions;
+    const { courseRunTypeOptions } = parsedTypeOptions;
+    const { entitlementUUIDS } = parsedTypeOptions;
```

Comment:
The linter allowed this? I thought our style was
```
    const {
        courseTypeOptions,
        courseRunTypeOptions,
        entitlementUUIDS,
    } = parsedTypeOptions;
```

That said, your style is less lines, but more characters.... hmmm

Reply:
It did allow it. Maybe it is better to deconstruct again though for consistency. I don't really have a preference. I'll see if one speaks to me